### DISCUSSION TITLE
Utility function to convert time series data for DBN training

### DISCUSSION THREAD

**Discussion by Maghnie** [pgmpy/pgmpy#1754] [Q&A]
Is there a convenience function somewhere in pgmpy that converts time series data (e.g. as shown in the pic below) into a format that's suitable for the [`fit` method](https://pgmpy.org/_modules/pgmpy/models/DynamicBayesianNetwork.html#DynamicBayesianNetwork.fit) of the dynamic bayesian network class?

![image](https://github.com/pgmpy/pgmpy/assets/68941589/e7e3e1f7-df07-4916-8d5c-acb9702e4823)
[Image source](https://www.researchgate.net/publication/333156822_An_Introduction_of_NoSQL_Databases_Based_on_Their_Categories_and_Application_Industries/figures?lo=1&utm_source=google&utm_medium=organic)

**Comment by ankurankan** [comment]
@Maghnie Sorry for the late reply. Looking at the data, I am not sure if Dynamic Bayesian Network (DBN) would be suitable for the analysis. The DBN model (and fitting) assumes that there are multiple observations at each time point which does not seem to be the case in the data that you have shared. I don't know what exactly is your goal with the analysis, but maybe you could get rid of the time variable and analyze the relationship between the other variables.

**Comment by Maghnie** [comment]
Thanks for the feedback @ankurankan — especially in an open-source project where the developers volunteer their time and resources, it's great that community questions are being regularly answered!

My goal is to perform prediction and anomaly detection on new data sets from the same building energy system, perhaps in real time as the live measurements come in.

I'm still learning about DBNs, so I'm not sure if there are other, more direct ways to train DBNs on time series data.
That's why my current idea is to start with what you mention at the end. I'd start by defining the number of time slices as T, then dividing the measurement rows into T groups and kind of transposing them into columns. 

If you have feedback about this approach, I'd appreciate the expert feedback! Thanks
